amplipi/streams.py

```diff
@@ -1321,6 +1323,216 @@ def info(self) -> models.SourceInfo:
     return source
 
 
+MUSIC_EXTENSIONS = ('.mp3', '.wav', '.aac', '.m4a', '.flac', '.aiff')
+
+
+class MediaDevice(PersistentStream, Browsable):
+  """ An external media device plugged into AmpliPro """
+
+  stream_type: ClassVar[str] = 'mediadevice'
+
+  def __init__(self, name: str, url: Optional[str], disabled: bool = False, mock: bool = False):
+    super().__init__(self.stream_type, name, disabled=disabled, mock=mock)
+    self.url = url
+    self.directory = '/media'
+    self.local_directory = '/media'
+    self.bkg_thread: Optional[threading.Thread] = None
+    self.supported_cmds = ['play', 'pause', 'next', 'prev']
+    self.song_list: List[str] = []
+    self.directory_list: List[str] = []
+    self.song_index = 0
+    self.ended = False
+    self._ended_timeout = datetime.datetime.now()
+    self.playing = ''
+    self.device: Optional[str] = None
+
+  def reconfig(self, **kwargs):
+    reconnect_needed = False
+    if 'disabled' in kwargs:
+      self.disabled = kwargs['disabled']
+    if 'name' in kwargs:
+      self.name = kwargs['name']
+    if 'url' in kwargs:
+      self.url = kwargs['url']
+      reconnect_needed = True
+    if reconnect_needed and self.is_activated():
+      self.restart()
+
+  def _activate(self, vsrc: int, remake_list: bool = True):
+    """ Connect a short run VLC process with audio output to a given audio source """
+    logger.info(f'connecting {self.name} to {vsrc}...')
+
+    if not self.mock and vsrc is not None:
```

Comment:
vsrc is known to be not none as declared by the function signature. Is there a reason to doubt it?